service/src/main/kotlin/io/provenance/digitalcurrency/consortium/frameworks/CoinMovementMonitor.kt

```diff
@@ -0,0 +1,86 @@
+package io.provenance.digitalcurrency.consortium.frameworks
+
+import io.provenance.digitalcurrency.consortium.api.CoinMovementRequest
+import io.provenance.digitalcurrency.consortium.api.CoinMovementRequestItem
+import io.provenance.digitalcurrency.consortium.bankclient.BankClient
+import io.provenance.digitalcurrency.consortium.config.CoinMovementProperties
+import io.provenance.digitalcurrency.consortium.config.EventStreamProperties
+import io.provenance.digitalcurrency.consortium.config.logger
+import io.provenance.digitalcurrency.consortium.domain.CoinMovementBookmarkRecord
+import io.provenance.digitalcurrency.consortium.domain.CoinMovementRecord
+import io.provenance.digitalcurrency.consortium.domain.EventStreamRecord
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.springframework.context.event.EventListener
+import org.springframework.stereotype.Component
+import java.util.UUID
+import kotlin.concurrent.thread
+
+@Component
+class CoinMovementMonitor(
+    coinMovementProperties: CoinMovementProperties,
+    eventStreamProperties: EventStreamProperties,
+    val bankClient: BankClient,
+) {
+    private val log = logger()
+
+    private val eventStreamId = UUID.fromString(eventStreamProperties.id)
```

Comment:
is this the same event stream id as the one rana is using? doesn't it have to be different?

Reply:
oh nvm i think i misunderstood what this was - this is just reading records to push

Reply:
We're using the same event stream reader. It just dispatches to both of our logic.